velox/functions/sparksql/RaiseError.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/functions/Macros.h"
+namespace facebook::velox::functions::sparksql {
+
+template <typename T>
+struct RaiseErrorFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      UnknownValue& result,
+      const arg_type<Varchar>& input) {
+    throw std::runtime_error(input);
```

Comment:
In Spark, exception is thrown for null input, while this implementation seems to give null output for null input. Could you confirm?

Reply:
yes, actually i didn't catch the case for null output,  could you share how to catch the case ?

Reply:
ping @rui-mo thanks for the above question  - how to process for null input case

Reply:
@gaoyangxiaozhu Sorry for the delayed response; I just got back from vacation.
Null input could be handled referring to https://facebookincubator.github.io/velox/develop/scalar-functions.html#null-behavior. I also notice https://github.com/facebookincubator/velox/commit/194772c823e73f3847ab1754a70bcda55a6c0679 added a similar Presto function. Is it possible to reuse?

Reply:
thanks, @rui-mo i think it is not ok to re-use, since `fail` throw `user error` while for spark we expect `runtime error` and also `fail` function return null if input is null but spark expect throw runtime exception with empty msg